sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2867,6 +2867,18 @@ pi_result cuda_piEnqueueKernelLaunch(
       retImplEv->start();
     }
 
+    // Set local mem max size if env var is present
+    {
+      const char *local_mem_sz_ptr = std::getenv("SYCL_PI_CUDA_MAX_LOCAL_MEM_SZ");
+      if (local_mem_sz_ptr) {
+        const int val = std::atoi(local_mem_sz_ptr);
```

Comment:
This part would be called fairly frequently so it would make sense to cache the read environment variable:

```suggestion
      static const char *local_mem_sz_ptr = std::getenv("SYCL_PI_CUDA_MAX_LOCAL_MEM_SZ");
      if (local_mem_sz_ptr) {
        static const int val = std::atoi(local_mem_sz_ptr);
```